pkg/core/transaction/witness.go

```diff
@@ -28,7 +28,7 @@ func (w *Witness) EncodeBinary(bw *io.BinWriter) {
 }
 
 // MarshalJSON implements the json marshaller interface.
-func (w *Witness) MarshalJSON() ([]byte, error) {
+func (w Witness) MarshalJSON() ([]byte, error) {
```

Comment:
Side note: https://github.com/golang/go/issues/22967
I really don't like non-pointer receiver methods, but so be it if it doesn't work otherwise.